lib/msal-node/src/client/ManagedIdentityApplication.ts

```diff
@@ -183,15 +224,38 @@ export class ManagedIdentityApplication {
 
             return cachedAuthenticationResult;
         } else {
-            // make a network call to the managed identity source
-            return this.managedIdentityClient.sendManagedIdentityTokenRequest(
+            return this.acquireTokenFromManagedIdentity(
                 managedIdentityRequest,
                 this.config.managedIdentityId,
                 this.fakeAuthority
             );
         }
     }
 
+    /**
+     * Acquires a token from a managed identity endpoint.
+     *
+     * @param managedIdentityRequest - The request object containing parameters for the managed identity token request.
+     * @param managedIdentityId - The identifier for the managed identity (e.g., client ID or resource ID).
+     * @param fakeAuthority - A placeholder authority used for the token request.
+     * @param refreshAccessToken - Optional flag indicating whether to force a refresh of the access token.
+     * @returns A promise that resolves to an AuthenticationResult containing the acquired token and related information.
+     */
+    private async acquireTokenFromManagedIdentity(
```

Comment:
Why is this method needed?

Reply:
I extracted the functionality for the network request since it's being used multiple times.

Reply:
This allowed me to write better docs for it